src/services/WorkspaceApi.ts

```diff
@@ -1,100 +1,33 @@
+export type WorkspaceApiConfig = PrefectConfig | CloudApiConfig
+
+export function isCloudConfig(config: WorkspaceApiConfig): config is CloudApiConfig {
   return 'accountId' in config && 'workspaceId' in config && 'token' in config
 }
 
-export type WorkspaceApiConfig = PrefectConfig | PrefectCloudConfig
-
-export class WorkspaceApi {
-
-  private readonly workspaceConfig: WorkspaceApiConfig
-
-  protected routePrefix: string = ''
-
-  public constructor(workspaceConfig: WorkspaceApiConfig) {
-    this.workspaceConfig = workspaceConfig
-  }
-
-  private get config(): AxiosRequestConfig {
-    return {
-      baseURL: this.baseUrl,
-      headers: this.headers,
-    }
-  }
-
-  private get baseUrl(): string {
-    const { baseUrl } = this.workspaceConfig
-
-    if (isCloudConfig(this.workspaceConfig)) {
-      const { accountId, workspaceId } = this.workspaceConfig
-
-      return `${baseUrl}/accounts/${accountId}/workspaces/${workspaceId}${this.routePrefix}`
-    }
-
-    return `${baseUrl}${this.routePrefix}`
-  }
-
-  private get headers(): AxiosRequestHeaders {
-    const headers: AxiosRequestHeaders = {
-      'X-PREFECT-UI': true,
-    }
-
-    if (isCloudConfig(this.workspaceConfig)) {
-      const { token } = this.workspaceConfig
-
-      headers.Authorization = `bearer ${token}`
-    }
-
-    return headers
-  }
-
-  protected instance(): AxiosInstance {
-    return axios.create(this.config)
-  }
-
-  protected get<T, R = AxiosResponse<T>>(url: string, config?: AxiosRequestConfig): Promise<R> {
-    return this.instance().get(url, config)
+export const getCloudBaseUrl: GetApiBaseUrl = (config) => {
+  if (!isCloudConfig(config)) {
+    return config.baseUrl
   }
 
-  protected delete<T, R = AxiosResponse<T>>(url: string, config?: AxiosRequestConfig): Promise<R> {
-    return this.instance().delete(url, config)
+  if (config.accountId && config.workspaceId) {
+    return `${config.baseUrl}/accounts/${config.accountId}/workspaces/${config.workspaceId}`
   }
 
-  protected head<T, R = AxiosResponse<T>>(url: string, config?: AxiosRequestConfig): Promise<R> {
-    return this.instance().head(url, config)
-  }
-
-  protected options<T, R = AxiosResponse<T>>(url: string, config?: AxiosRequestConfig): Promise<R> {
-    return this.instance().options(url, config)
-  }
-
-  // axios uses any for the data argument
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  protected post<T, R = AxiosResponse<T>>(url: string, data?: any, config?: AxiosRequestConfig): Promise<R> {
-    return this.instance().post(url, data, config)
-  }
+  throw new CloudConfigMissingParamsError()
+}
 
-  // axios uses any for the data argument
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  protected put<T, R = AxiosResponse<T>>(url: string, data?: any, config?: AxiosRequestConfig): Promise<R> {
-    return this.instance().put(url, data, config)
-  }
+export class WorkspaceApi extends BaseApi<WorkspaceApiConfig> {
```

Comment:
Github won't let me do a suggestion for this whole file. But couple things I here. 

1. You've defined and exported `getCloudBaseUrl`. But its also used for OSS so I think `getWorkspaceBaseUrl` is more accurate (also follows the naming convention of this file). I'm guessing that naming is just left over from when the only api like this was the workspace api and not really a conscious decision you made. 
2. You've created a constructor (which you had to type and call super in) just to set `this.getBaseUrl`. Which is `protected` and can just be an override directly in this class (same result).
3. I personally wouldn't abstract or export the `getCloudBaseUrl`. Its not used anywhere else and its pretty specific to this file. If you needed that logic you probably should be extended WorkspaceApi directly. 
```typescript
export class WorkspaceApi extends BaseApi<WorkspaceApiConfig> {

  protected override getBaseUrl: GetApiBaseUrl = (config) => {
    if (!isCloudConfig(config)) {
      return config.baseUrl
    }

    if (config.accountId && config.workspaceId) {
      return `${config.baseUrl}/accounts/${config.accountId}/workspaces/${config.workspaceId}`
    }

    throw new CloudConfigMissingParamsError()
  }

}
```

Reply:
agreed with all of this